tests/tests_pytorch/strategies/test_xla.py

```diff
@@ -34,7 +37,7 @@ def on_train_start(self) -> None:
 def test_model_tpu_one_core():
     """Tests if device/debug flag is set correctly when training and after teardown for XLAStrategy."""
     model = BoringModelTPU()
-    trainer = Trainer(accelerator="tpu", devices=1, fast_dev_run=True, strategy=XLAStrategy(debug=True))
+    trainer = Trainer(fast_dev_run=True, strategy=XLAStrategy(debug=True))
```

Comment:
How many cores will the trainer select here?

Reply:
All, 4 in a v4-8

Reply:
The reason why I asked is because the test is named "one_core". Is relying on the auto-device count the desired behavior of this test is what I'm wondering.

Reply:
None of the assertions rely on running in a single core. It was probably done like this to avoid spawning expensive processes (similarly to how we regularly use ddp_spawn with 1 device on CPU/GPU). I'll update the test name